support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/TicketSerializer.java

```diff
@@ -0,0 +1,56 @@
+package org.apereo.cas;
+
+import org.apereo.cas.ticket.ServiceTicket;
+import org.apereo.cas.ticket.Ticket;
+import org.apereo.cas.ticket.TicketGrantingTicket;
+
+/**
+ * This is {@link TicketSerializer}.
+ *
+ * Abstraction layer for ticket serialization methods
+ *
+ * @author David Rodriguez
+ *
+ * @param <T> should be something that can be written
+ * to Cassandra like {@link String}
+ *
+ * @since 5.1.0
+ */
+public interface TicketSerializer<T> {
```

Comment:
The support module here, should try and make every effort possible to NOT know anything about service tickets, TGTs, etc. Not in the API, not in method names, not anywhere as much as possible.  It should try to work with a very generic ticket abstraction and then learn as much as it can at runtime from the ticket catalog.